tests/phpunit/tests/post/query.php

```diff
@@ -777,4 +777,525 @@ public function test_found_posts_should_be_integer_even_if_found_posts_filter_re
 
 		$this->assertIsInt( $q->found_posts );
 	}
+
+	/**
+	 * @ticket 47280
+	 */
+	public function test_found_posts_are_correct_for_empty_query() {
+		self::factory()->post->create_many( 12 );
+
+		$q = new WP_Query();
+
+		$this->assertSame( 0, $q->post_count );
+		$this->assertSame( 0, $q->found_posts );
+		$this->assertSame( 0, $q->max_num_pages );
+	}
+
+	/**
+	 * @ticket 47280
+	 * @dataProvider data_fields
+	 *
+	 * @param string $fields Value of the `fields` argument for `WP_Query`.
+	 */
+	public function test_found_posts_are_correct_for_query_with_no_results( $fields ) {
+		$q = new WP_Query(
+			array(
+				'fields'       => $fields,
+				'posts_status' => 'draft',
+			)
+		);
+
+		$this->assertSame( 0, $q->post_count, self::get_count_message( $q ) );
+		$this->assertSame( 0, $q->found_posts, self::get_count_message( $q ) );
+		$this->assertSame( 0, $q->max_num_pages, self::get_count_message( $q ) );
+	}
+
+	/**
+	 * @ticket 47280
+	 * @dataProvider data_fields
+	 *
+	 * @param string $fields Value of the `fields` argument for `WP_Query`.
+	 */
+	public function test_found_posts_are_correct_for_basic_query( $fields ) {
+		self::factory()->post->create_many( 5 );
+
+		$q = new WP_Query(
+			array(
+				'fields'         => $fields,
+				'posts_per_page' => 2,
+			)
+		);
+
+		$this->assertSame( 2, $q->post_count, self::get_count_message( $q ) );
+		$this->assertSame( 5, $q->found_posts, self::get_count_message( $q ) );
+		$this->assertEquals( 3, $q->max_num_pages, self::get_count_message( $q ) );
+	}
+
+	/**
+	 * @ticket 47280
+	 * @dataProvider data_fields
+	 *
+	 * @param string $fields Value of the `fields` argument for `WP_Query`.
+	 */
+	public function test_found_posts_are_correct_for_query_with_no_limit( $fields ) {
+		self::factory()->post->create_many( 5 );
+
+		$q = new WP_Query(
+			array(
+				'fields'         => $fields,
+				'posts_per_page' => -1,
+			)
+		);
+
+		$this->assertSame( 5, $q->post_count, self::get_count_message( $q ) );
+		$this->assertSame( 5, $q->found_posts, self::get_count_message( $q ) );
+		// You would expect this to be 1 but historically it's 0 for posts without paging.
+		$this->assertSame( 0, $q->max_num_pages, self::get_count_message( $q ) );
+	}
+
+	/**
+	 * @ticket 47280
+	 * @dataProvider data_fields
+	 *
+	 * @param string $fields Value of the `fields` argument for `WP_Query`.
+	 */
+	public function test_found_posts_are_correct_for_query_with_no_paging( $fields ) {
+		self::factory()->post->create_many( 5 );
+
+		$q = new WP_Query(
+			array(
+				'fields'   => $fields,
+				'nopaging' => true,
+			)
+		);
+
+		$this->assertSame( 5, $q->post_count, self::get_count_message( $q ) );
+		$this->assertSame( 5, $q->found_posts, self::get_count_message( $q ) );
+		// You would expect this to be 1 but historically it's 0 for posts without paging.
+		$this->assertSame( 0, $q->max_num_pages, self::get_count_message( $q ) );
+	}
+
+	/**
+	 * @ticket 47280
+	 * @dataProvider data_fields
```

Comment:
```suggestion
	 * @ticket 47280
	 *
	 * @dataProvider data_fields
```

It seems likely requiring a space will be introduced in the near future. See the second code block in the description of [WP#56070](https://core.trac.wordpress.org/ticket/56070)

🔢 Note: This comment applies in multiple locations but I've only noted it once.